react/MuiCozyTheme/Menus/index.js

```diff
@@ -1,3 +1,122 @@
-import Menu from '@material-ui/core/Menu'
+import React, { Component } from 'react'
 
-export default Menu
+import MenuButton from '../../Button'
+import PropTypes from 'prop-types'
+
+import ClickAwayListener from '@material-ui/core/ClickAwayListener'
+import Grow from '@material-ui/core/Grow'
+import Popper from '@material-ui/core/Popper'
+import Paper from '@material-ui/core/Paper'
+import MenuList from '@material-ui/core/MenuList'
+import { withStyles } from '@material-ui/core/styles'
+
+const styles = {
+  paper: {
+    '& hr': {
+      margin: '8px 0',
+      border: 0,
+      borderTop: '1px solid var(--silver)'
+    }
+  }
+}
+
+class Menu extends Component {
+  state = { open: false, anchorEl: null }
+
+  toggle = event => {
+    const { currentTarget } = event
+    this.setState(state => ({
+      anchorEl: currentTarget,
+      open: !state.open
+    }))
+  }
+
+  close = () => {
+    this.setState({
+      open: false
+    })
+  }
+
+  render() {
+    const {
+      label,
+      disabled,
+      buttonClassName,
+      component,
+      children,
+      placement,
+      classes
+    } = this.props
+    const { open, anchorEl } = this.state
+    const offset = {
+      offset: {
+        offset: '0, 4'
+      }
+    }
+
+    return (
+      <ClickAwayListener onClickAway={this.close}>
+        <div>
+          {!component ? (
+            <MenuButton
+              disabled={disabled}
+              onClick={this.toggle}
+              label={label}
+              buttonClassName={buttonClassName}
+            />
+          ) : (
+            React.cloneElement(component, {
+              disabled,
+              onClick: this.toggle
+            })
+          )}
+          {
+            <Popper
+              open={open}
+              anchorEl={anchorEl}
+              transition
+              placement={placement}
+              modifiers={offset}
+            >
+              {({ TransitionProps, placement }) => (
+                <Grow
+                  {...TransitionProps}
+                  style={{
+                    transformOrigin:
+                      placement === 'bottom-end'
+                        ? 'center top'
+                        : 'center bottom'
+                  }}
+                >
+                  <Paper className={classes.paper}>
+                    <MenuList>{children}</MenuList>
+                  </Paper>
+                </Grow>
+              )}
+            </Popper>
+          }
+        </div>
+      </ClickAwayListener>
+    )
+  }
+}
+
+Menu.defaultProps = {
+  disabled: false,
+  component: null,
+  placement: 'bottom-end'
+}
+
+Menu.propTypes = {
+  /** Disables the menu */
+  disabled: PropTypes.bool,
+  /** Specifies a custom component for the opener */
+  component: PropTypes.element,
+  /** placement for the popover */
+  placement: PropTypes.oneOf(['bottom-start', 'bottom-end'])
```

Comment:
I think we should keep `transformOrigin` syntax, it is known by everybody doing CSS and it is more versatile. It also maps directly to the Grow interface.